upup/pkg/fi/nodeup/nodetasks/bootstrap_client.go

```diff
@@ -155,6 +159,19 @@ func (b *KopsBootstrapClient) QueryBootstrap(ctx context.Context, req *nodeup.Bo
 		}
 	}
 
+	if ips, err := net.LookupIP(b.BaseURL.Host); err != nil {
+		if dnsErr, ok := err.(*net.DNSError); ok && dnsErr.IsNotFound {
+			return nil, fi.NewTryAgainLaterError("kops-controller DNS not setup yet")
+		}
+		return nil, err
+	} else {
+		for _, ip := range ips {
+			if ip.String() == cloudup.PlaceholderIP {
+				return nil, fi.NewTryAgainLaterError("kops-controller DNS not setup yet")
```

Comment:
Probably also a good idea to differentiate the two messages e.g. ("kops-controller DNS not setup yet (placeholder IP found)" vs "kops-controller DNS not setup yet (error was %v)"

Reply:
Good idea, it turns out `URL.Host` includes the port number which will never resolve. Updating it to `URL.Hostname()` seems to have shaved off a couple minutes on the `kops validate cluster` time in this small sample size of presubmit jobs.